topology/intel/dmic/dmic-macros.h

```diff
@@ -0,0 +1,262 @@
+#define FIR_CONTROL_A_ARRAY_START_EN(x)		SET_BIT(6, x)
+#define FIR_CONTROL_A_DCCOMP(x)			SET_BIT(4, x)
+#define FIR_CONTROL_A_MUTE(x)			SET_BIT(1, x)
+#define FIR_CONTROL_A_STEREO(x)			SET_BIT(0, x)
+
+/* FIR_CONFIG_A bits */
+#define FIR_CONFIG_A_FIR_DECIMATION(x)		SET_BITS(20, 16, x)
+#define FIR_CONFIG_A_FIR_SHIFT(x)		SET_BITS(11, 8, x)
+#define FIR_CONFIG_A_FIR_LENGTH(x)		SET_BITS(7, 0, x)
+
+/* DC offset compensation time constants */
+#define DCCOMP_TC0	0
+#define DCCOMP_TC1	1
+#define DCCOMP_TC2	2
+#define DCCOMP_TC3	3
+#define DCCOMP_TC4	4
+#define DCCOMP_TC5	5
+#define DCCOMP_TC6	6
+#define DCCOMP_TC7	7
+
+/* DC_OFFSET_LEFT_A bits */
+#define DC_OFFSET_LEFT_A_DC_OFFS(x)		SET_BITS(21, 0, x)
+
+/* DC_OFFSET_RIGHT_A bits */
+#define DC_OFFSET_RIGHT_A_DC_OFFS(x)		SET_BITS(21, 0, x)
+
+/* OUT_GAIN_LEFT_A bits */
+#define OUT_GAIN_LEFT_A_GAIN(x)			SET_BITS(19, 0, x)
+
+/* OUT_GAIN_RIGHT_A bits */
+#define OUT_GAIN_RIGHT_A_GAIN(x)		SET_BITS(19, 0, x)
+
+/* FIR_CONTROL_B bits */
+#define FIR_CONTROL_B_START_BIT			BIT(7)
+#define FIR_CONTROL_B_ARRAY_START_EN_BIT	BIT(6)
+#define FIR_CONTROL_B_MUTE_BIT			BIT(1)
+#define FIR_CONTROL_B_START(x)			SET_BIT(7, x)
+#define FIR_CONTROL_B_ARRAY_START_EN(x)		SET_BIT(6, x)
+#define FIR_CONTROL_B_DCCOMP(x)			SET_BIT(4, x)
+#define FIR_CONTROL_B_MUTE(x)			SET_BIT(1, x)
+#define FIR_CONTROL_B_STEREO(x)			SET_BIT(0, x)
+
+/* FIR_CONFIG_B bits */
+#define FIR_CONFIG_B_FIR_DECIMATION(x)		SET_BITS(20, 16, x)
+#define FIR_CONFIG_B_FIR_SHIFT(x)		SET_BITS(11, 8, x)
+#define FIR_CONFIG_B_FIR_LENGTH(x)		SET_BITS(7, 0, x)
+
+/* DC_OFFSET_LEFT_B bits */
+#define DC_OFFSET_LEFT_B_DC_OFFS(x)		SET_BITS(21, 0, x)
+
+/* DC_OFFSET_RIGHT_B bits */
+#define DC_OFFSET_RIGHT_B_DC_OFFS(x)		SET_BITS(21, 0, x)
+
+/* OUT_GAIN_LEFT_B bits */
+#define OUT_GAIN_LEFT_B_GAIN(x)			SET_BITS(19, 0, x)
+
+/* OUT_GAIN_RIGHT_B bits */
+#define OUT_GAIN_RIGHT_B_GAIN(x)		SET_BITS(19, 0, x)
+
+/* FIR coefficients */
+#define FIR_COEF_A(x)				SET_BITS(19, 0, x)
+#define FIR_COEF_B(x)				SET_BITS(19, 0, x)
+
+/* Minimum OSR is always applied for 48 kHz and less sample rates */
+#define DMIC_MIN_OSR  50
+
+/* These are used as guideline for configuring > 48 kHz sample rates. The
+ * minimum OSR can be relaxed down to 40 (use 3.84 MHz clock for 96 kHz).
+ */
+#define DMIC_HIGH_RATE_MIN_FS	64000
+#define DMIC_HIGH_RATE_OSR_MIN	40
+
+/* Used for scaling FIR coefficients for HW */
+#define DMIC_HW_FIR_COEF_MAX ((1 << (DMIC_HW_BITS_FIR_COEF - 1)) - 1)
+#define DMIC_HW_FIR_COEF_Q (DMIC_HW_BITS_FIR_COEF - 1)
+
+/* Internal precision in gains computation, e.g. Q4.28 in int32_t */
+#define DMIC_FIR_SCALE_Q 28
+
+/* Used in unmute ramp values calculation */
+#define DMIC_HW_FIR_GAIN_MAX ((1 << (DMIC_HW_BITS_FIR_GAIN - 1)) - 1)
+
+/* Hardwired log ramp parameters. The first value is the initial gain in
+ * decibels. The second value is the default ramp time.
+ */
+#define LOGRAMP_START_DB Q_CONVERT_FLOAT(-90, DB2LIN_FIXED_INPUT_QY)
```

Comment:
Sorry I didn't notice in earlier round, the LOGRAMP #defines are unnecessary.

Reply:
ok